internal/docstore/mongodocstore/mongo.go

```diff
@@ -454,6 +488,166 @@ func (c *collection) docID(doc driver.Document) (interface{}, error) {
 	return id, nil
 }
 
+type indexedAction struct {
+	index  int
+	action *driver.Action
+}
+
+type indexedError = struct {
+	Index int
+	Err   error
+}
+
+func (c *collection) runActionsUnordered(ctx context.Context, actions []*driver.Action) driver.ActionListError {
+	// MongoDB has a bulk write RPC, but no easy way to do a bulk get.
+	// We run the Gets in parallel.
+	// TODO(jba): consider doing a bulk get by using Find with an "or" filter.
+	var alerr driver.ActionListError
+
+	var gets, writes []indexedAction
+	for i, a := range actions {
+		if a.Kind == driver.Get {
+			gets = append(gets, indexedAction{i, a})
+		} else {
+			writes = append(writes, indexedAction{i, a})
+		}
+	}
+
+	// Run each Get action concurrently.
+	getc := make(chan indexedError)
+	for _, g := range gets {
+		g := g
+		go func() {
+			err := c.get(ctx, g.action)
+			getc <- indexedError{g.index, err}
+		}()
+	}
+
+	// Run all writes in a single BulkWrite RPC.
+	alerrBulk := c.unorderedBulkWrite(ctx, writes)
+	if alerrBulk != nil {
+		alerr = append(alerr, alerrBulk...)
```

Comment:
How about doing something similar to the firestore one, make a slice here and give a proper length to the error list?

Reply:
The problem is how to handle the errors that aren't associated with any action (the ones with -1).